toponetx/algorithms/distance.py

```diff
@@ -0,0 +1,147 @@
+"""Module to distance between nodes or cells on topological domains."""
+from collections.abc import Collection, Hashable, Iterable, Iterator
+from warnings import warn
+
+import networkx as nx
+import numpy as np
+
+from toponetx.classes.cell import Cell
+from toponetx.classes.cell_complex import CellComplex
+from toponetx.classes.colored_hypergraph import ColoredHyperGraph
+from toponetx.classes.combinatorial_complex import CombinatorialComplex
+from toponetx.classes.hyperedge import HyperEdge
+from toponetx.classes.simplicial_complex import SimplicialComplex
+from toponetx.exception import TopoNetXError
+
+__all__ = ["distance", "cell_distance"]
+
+
+def distance(complex, source, target, s=1):
+    """Return shortest s-walk distance between two nodes in the cell complex.
+
+    Parameters
+    ----------
+    complex : Complex, supported complexes are cell/combintorial and hypegraphs
+    source : a hashable represeting a node in the complex
+        A node in the input complex.
+    target : a hashable represeting a node
+        A node in the input complex.
+    s : int
+        the number of intersections between pairwise consecutive cells
+
+    Returns
+    -------
+    s_walk_distance : int
```

Comment:
single return values do not have a name. Remove the s_connected_components : part (same throughout this file).